### ISSUE TITLE
Dialog headline title

### ISSUE THREAD

**Issue by bembelimen** [joomla/joomla-cms#45429]
### Steps to reproduce the issue
1.  Open any dialog (e.g. during the installation changing the language or the batch modal in com_content backend)
2. Inspect the dialog/headline with your browser developer tool

### Expected result
- Dialog has ```aria-labelledby="xxx"``` pointing to the headline of the dialog 

### Actual result
- No pointers to the headline

### Additional comments

Issue was mentioned by @exlemor and confirmed by @brianteeman [in this PR](https://github.com/joomla/joomla-cms/pull/44735#issuecomment-2837204127), so probably you have an idea how to solve this?

Current status is: Dialogs are generated in [this file as web component](https://github.com/joomla/joomla-cms/blob/6.0-dev/build/media_source/system/js/joomla-dialog.w-c.es6.js). Specific in [line 188](https://github.com/joomla/joomla-cms/blob/6.0-dev/build/media_source/system/js/joomla-dialog.w-c.es6.js#L188) the dialog element is created. I think the attributes has to be added when there is a header in [line 206](https://github.com/joomla/joomla-cms/blob/6.0-dev/build/media_source/system/js/joomla-dialog.w-c.es6.js#L206).
Now here is the challenge: the headline needs an ID, which is not that easy, especially if there are more than one dialog at the page. Do we need to remember how many times a dialog is created?

Any idea?

//cc @Fedik as you know the dialog I guess :-)

**Comment by Fedik** [comment]
This need a clarification, there nothing about it at https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/dialog documentation

**Comment by brianteeman** [comment]
https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/

https://dequeuniversity.com/library/aria/simple-dialog

**Comment by brianteeman** [comment]
it is a requirement

**Comment by Fedik** [comment]
No one said that, in the linked articles about `dialog`  (NOT `div role=dialog`) element.

But we can add.

**Comment by brianteeman** [comment]
it doesnt say that having an alt description on an image is a requirement in the specification of the img element either but it is still an accessibility requirement
:)

**Comment by Fedik** [comment]
Yeah, that is true 😄

**Comment by chmst** [comment]
The dialog headline has a non-tranlated "JCLOSE" for the dismiss button. 

![Image](https://github.com/user-attachments/assets/2b398612-aa8c-46d3-9ee2-3b9ef9c8a128)

**Comment by Fedik** [comment]
It is missing `JText::script('JCLOSE');` somewhere

**Comment by brianteeman** [comment]
Language string addressed in #45520

**Comment by Fedik** [comment]
There it is 
- https://github.com/joomla/joomla-cms/pull/46089
